File: jexler-war/src/main/webapp/WEB-INF/jexlers/cron.py
# autostart
import time
from net.jexler import StopService
from net.jexler import CronService

services.add(CronService(jexler, "every-minute").setCron("* * * * *").start())
services.add(CronService(jexler, "every-two-minutes").setCron("*/2 * * * *").start())

while True:
    event = events.take()
    if isinstance(event, CronService.Event):
        now = time.strftime("%b %d %Y %H:%M:%S", time.localtime())
        print "Is is now (jython): %s (%s)" % (now, event.getServiceId())
    elif isinstance(event, StopService.Event):
        break

File: jexler-war/src/main/webapp/WEB-INF/jexlers/cron.rb
# autostart
import Java::net.jexler.StopService
import Java::net.jexler.CronService
import Java::net.jexler.ShellTool

$services.add CronService.new($jexler, "every-minute").setCron("* * * * *").start
$services.add CronService.new($jexler, "every-two-minutes").setCron("*/2 * * * *").start

begin
  event = $events.take
  if event.is_a? CronService::Event
    puts "It is now (ruby) " + Time.new.inspect + " (" + event.getServiceId + ")"

    shellTool = ShellTool.new
    result = shellTool.run "echo 'hello world'"
    puts "rc = " + result.rc.to_s
    puts "stdout = " + result.stdout
    puts "stderr = " + result.stderr

    #$log.debug "hello"
    raise "something goes wrong"

  elsif event.is_a? StopService::Event
    return
  end
end while true
